### ISSUE TITLE
`class-naming` の値を `""` または `[""]` にした場合にwarningが表示される

### ISSUE THREAD

**Issue by kubosho** [markuplint/markuplint#1976]
- Markuplint version: `4.9.2`
- Parser lang: Astro, HTML
- Node.js version: `20.11.1`
- OS: Windows

## Describe the bug

[class-naming](https://markuplint.dev/ja/docs/rules/class-naming)の値を `''` にした場合、下記のようにパターンとマッチしない旨のwarningが表示されます。

```shell
<markuplint> warning: クラス名「PublishedDate」は次のパターンにマッチしませんでした "" (class-naming)
~/src/github.com/kubosho/blog.kubosho.com/src/app/components/EntryList.astro:51:22
50: ••••••<li•class="EntryItem">
51: ••••••••<time•class="PublishedDate"•datetime={formatIsoString(entry.data.publishedAt)}>
52: ••••••••••{formatYYMDString(entry.data.publishedAt,•{•year:•'年',•month:•'月',•day:•'日'•})}"
```

Playgroundで `class-naming` の値を `""` または `[""]` にした場合も同じ挙動でした。

## Playground URL

https://playground.markuplint.dev/#N4Igxg9gdgZglgcxALhMAOlABF9ICmAHgC75QAmAznslgNp4C2AhgE4DWArgA4A2cUYslb5IjRmXL5yeALoAaTDjytOvfNRRYM2HLnC9mlSgFoozRgIQ19eJVgC+mJ1BDzwEKQDE46gCoAntz4KCAAdAAWxIy8bh5SoTgAPORwAG5YYIbGALx4YMykCBCsAXgAfPY4AMKF+MWlAIz2SQD0qWmVrg5AA

## Steps To Reproduce

1. `"rules": { "class-naming": "" }` をMarkuplintの設定ファイルに書く
2. HTMLファイルなどでクラス名を定義する

## The current behavior

`class-naming` の値を `''` または `[]` にした場合に、Markuplintを実行したときにwarningが表示される。

## The expected behavior

`class-naming` の値を `''` または `[]` にした場合は、Markuplintを実行したときにwarningが表示されない。

または[class-namingのドキュメント](https://markuplint.dev/ja/docs/rules/class-naming)のInterfaceが、[ソースコード](https://github.com/markuplint/markuplint/blob/3dc5aa4/packages/%40markuplint/rules/src/class-naming/index.ts#L8)と同じく `string | string[] | null` に修正される。

**Comment by kubosho** [comment]
クラス名の値を空文字 (`class=""`) にした場合はwarningが表示されないことを確認しています。

**Comment by YusukeHirao** [comment]
@kubosho いつもありがとうございます。

`"class-naming": ""`とした場合の期待ですが、これは「空文字にマッチする」という意味になってしまいます。

`class-naming`自体を無効化する（もしくは何でも許容する）場合は`null`が設定値で、たしかにドキュメントのIntefaceに記載はないのですが、他のルールもすべて`null`は無効化を意味します。

`class-naming`の値は
- 文字列の場合は完全マッチ
- `/`で囲った文字列（`"class-naming": "/^foo-.+$/"`のような形式）は正規表現扱い
となります。
なので「何でも許容する」を敢えて設定値を作るとしたら

```json
"class-naming": "/.*/"
```

でもよいです。

**Comment by kubosho** [comment]
回答ありがとうございます！
